airmon-ng.linux: in a for loop, replace `ls` with a glob

wileyhy opened the pull request:
The `[ -e` syntax covers the case of when the glob produces no matches and `sh` uses the glob pattern itself as a value for "$i".

### xtrace comparison:

#### original
```sh
set -x; 
for i in $(ls /sys/class/ieee80211/); do 
  if [ ! -d "/sys/class/ieee80211/${i}/device/net" ]; then 
    MISSING_INTERFACE="${i}"; 
    echo "$MISSING_INTERFACE" :: "$i"; 
  fi; 
done; 
set - 
```

> \++ ls --color=auto /sys/class/ieee80211/
> \+ for i in $(ls /sys/class/ieee80211/)
> \+ '[' '!' -d /sys/class/ieee80211/phy0/device/net ']'
> \+ set -
> 

#### change added
```sh
set -x; 
for i in /sys/class/ieee80211/*; do 
  if [ -e "${i}" ] && [ ! -d "${i}/device/net" ]; then 
    MISSING_INTERFACE="${i}"; 
    echo "$MISSING_INTERFACE" :: "$i"; 
  fi; 
done; 
set -
```

> \+ for i in /sys/class/ieee80211/*
> \+ '[' -e /sys/class/ieee80211/phy0 ']'
> \+ '[' '!' -d /sys/class/ieee80211/phy0/device/net ']'
> \+ set -
> 

Edit 1: formatting
Edit 2: note re `[ -e` syntax

wileyhy committed:
airmon-ng.linux: in a for loop, replace `ls` with a glob

xtrace comparison:

# original
[root@localhost-live bug_airmon-ng-email]# set -x; for i in $(ls /sys/class/ieee80211/); do if [ ! -d "/sys/class/ieee80211/${i}/device/net" ]; then echo "$MISSING_INTERFACE" :: "$i"; fi; done; set -
++ ls --color=auto /sys/class/ieee80211/
+ for i in $(ls /sys/class/ieee80211/)
+ '[' '!' -d /sys/class/ieee80211/phy0/device/net ']'
+ set -

# change added
[root@localhost-live bug_airmon-ng-email]# set -x; for i in /sys/class/ieee80211/*; do if [ -e "${i}" ] && [ ! -d "${i}/device/net" ]; then MISSING_INTERFACE="${i}"; echo "$MISSING_INTERFACE" :: "$i"; fi; done; set -
+ for i in /sys/class/ieee80211/*
+ '[' -e /sys/class/ieee80211/phy0 ']'
+ '[' '!' -d /sys/class/ieee80211/phy0/device/net ']'
+ set -

ZeroChaos- commented:
This is a good start, but unfortunately it entirely breaks the function
```
+ iw phy /sys/class/ieee80211/phy0 interface add wlan0mon type monitor
+ IW_ERROR='command failed: No such file or directory (-2)' 
+ '[' -z 'command failed: No such file or directory (-2)' ']'
+ printf '\n\n ERROR adding %s mode interface: %s\n' monitor 'command failed: No such file or directory (-2)'
```

Also, please remove the two comments above your change since the comments are suggesting to do exactly what you are trying to do :-)

ZeroChaos- reviewed (changes requested):
Please fix setting MISSING_INTERFACE to the expected value.  You changed the meaning of ${i} but didn't update it's use everywhere

wileyhy committed:
airmon-ng.linux: lines 206-7, remove comments

Comments had read...

    #this should be fixed, but it's not going to be right now
    # shellcheck disable=2045

...which referred to...

https://www.shellcheck.net/wiki/SC2045

wileyhy committed:
airmon-ng.linux: Updated value of MISSING_INTERFACE

Fix setting MISSING_INTERFACE to the expected value, from commit e833818.

wileyhy commented:
You're right; thank you. Two fixes have commits up for review.

ZeroChaos- reviewed (approved):
tested on zsh, dash, bash, and mksh

Looks great, thanks!

ZeroChaos- commented:
merging, one unrelated check failure

ZeroChaos- commented:
thanks @wileyhy

wileyhy commented:
You're welcome! Thank _you_! :-)

On Wed, Jul 19, 2023, 11:59 AM Zero_Chaos ***@***.***> wrote:

> thanks @wileyhy <https://github.com/wileyhy>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/aircrack-ng/aircrack-ng/pull/2551#issuecomment-1642599596>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUF2F253NEEOA2FBBJJONQTXRAVBRANCNFSM6AAAAAA2D2HGQQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
